routers/web/repo/compare.go

```diff
@@ -265,7 +263,8 @@ func ParseCompareInfo(ctx *context.Context) *common.CompareInfo {
 				}
 				return nil
 			}
-			ci.HeadBranch = headInfos[1]
+			// FIXME: how to correctly choose the head repository? The logic below (3-8) is quite complex, the real head repo is determined there
```

Comment:
When trying to understand the logic, I find something strange, so I left a FIXME here.

Do you have ideas?

Reply:
I don't think so.

1. The new logic is inconsistent with the steps 3-8 below. The "headRepo" guess is quite complex.
2. I do not think GetRepoRawDiffForFile would work for different base&head repositories, for example: `ref1...headRepo:ref2`, but what if ref1 only exists in base repo and ref2 only exists in head repo?